MediaPlayer/MediaPlayer.js

```diff
@@ -391,16 +489,133 @@ const MediaPlayerBehaviorDecorator = hoc((config, Wrapped) => { // eslint-disabl
 		 * @memberof agate/MediaPlayer.MediaPlayerBase.prototype
 		 * @public
 		 */
-		loopChange = () => {
+		handleOnRepeat = () => {
+			// Handling the 3 states of repeat: repeat none, repeat one and repeat all
+			switch (this.state.repeat) {
+				case 'none':
+					this.setState(prevState  => {
+						return ({loop: !prevState.loop});
+					}, () => {
+						this.media.loop = this.state.loop;
+					});
+					this.setState({repeat: 'one'});
+					break;
+				case 'one':
+					this.setState({repeat: 'all'});
+					this.media.loop = false;
+					break;
+				case 'all':
+					this.setState({repeat: 'none'});
+					this.media.loop = false;
+			}
+		};
+
+		handleOnEnded = () => {
+			// Play next media only if current media is not last or, if it is last, only if repeat='all'
+			if (this.state.sourceIndex !== this.state.playlist.length - 1 || (this.state.sourceIndex === this.state.playlist.length - 1 && this.state.repeat === 'all')) {
+				this.handleNext();
+			}
+		};
+
+		handleNext = () => {
+			let currentIndex = this.state.sourceIndex;
+
+			if (currentIndex < this.state.playlist.length - 1) {
+				++currentIndex;
+			} else {
+				// When shuffle is true, the playback of the list restarts and the media list is reshuffled.
+				currentIndex = 0;
+
+				if (this.state.shuffle) {
+					this.shufflePlaylist();
+				}
+			}
+
+			this.setState(() => {
+				return ({sourceIndex: currentIndex});
+			}, () => {
+				this.play();
+			});
+		};
+
+		handlePrevious = () => {
+			let currentIndex = this.state.sourceIndex;
+
+			if (currentIndex > 0) {
```

Comment:
Shouldn't we check the media is playing, if so, go to the start of the current playing media?
WDYT?

Reply:
@stanca-pop-lgp Did you consider this comment?
When I click the previous button, it goes to the previous media instead of the beginning of the current playing media.